### ISSUE TITLE
Timeout on Linux architecture but not on Windows.

### ISSUE THREAD

**Issue by IngoVals** [docker/for-win#1688]
### Expected behavior
Pull the image succesfully on both container formats

### Actual behavior
Only able to pull when using Windows containers.

### Information and steps to reproduce the behavior

No proxy
DNS set to 8.8.8.8

_On Linux containers_

```
$ docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:12:29 2017
  OS/Arch:      linux/amd64
  Experimental: true
```

```
$ docker run hello-world
Unable to find image 'hello-world:latest' locally
C:\Program Files\Docker\Docker\Resources\bin\docker.exe: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'C:\Program Files\Docker\Docker\Resources\bin\docker.exe run --help'.
```

_Switch to Windows containers_

```
$ docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.24)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:15:52 2017
  OS/Arch:      windows/amd64
  Experimental: true
```

```
$ docker run hello-world

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (windows-amd64)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run a Windows Server container with:
 PS C:\> docker run -it microsoft/windowsservercore powershell

Share images, automate workflows, and more with a free Docker ID:
 https://cloud.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/engine/userguide/
```

**Comment by IngoVals** [comment]
Sorry about the mess, accidentally open the issue using `docker ps` when I thought I was focused in terminal. Decided to use that issue instead of closing and creating a new one.

**Comment by IngoVals** [comment]
I have found out that this is a known vpnkit bug but seems hard to find using google. This has been fixed but not released. Installing the edge version of docker for windows fixed this for me 

see:
https://github.com/Azure/azure-iot-pcs-remote-monitoring-dotnet/issues/62
https://github.com/docker/for-win/issues/1581
https://github.com/docker/for-win/issues/1432

This might be unrelated to the original issue here which I guess was a proxy issue but since google seems to point towards this post I think it's nice to leave this here at least.

**tl;dr** Uninstall _docker for windows_, install _docker for windows edge_ version. If you have images and containers [there are ways to save them between uninstall/install. ](https://docs.docker.com/docker-for-windows/faqs/#what-kind-of-feedback-are-we-looking-for)

**Comment by jasonbivins** [comment]
@IngoVals  
I resolved this same error today by changing the DNS settings from the manual 8.8.8.8 to automatic
The public google DNS was being blocked by corporate proxy.
